packages/terra-form/tests/nightwatch/number-field/number-field-spec.js

```diff
@@ -0,0 +1,48 @@
+/* eslint import/no-extraneous-dependencies: ["error", {"devDependencies": true}] */
+
+const screenshot = require('terra-toolkit').screenshot;
+
+module.exports = {
+  before: (browser, done) => {
+    browser.resizeWindow(browser.globals.width, browser.globals.height, done);
+  },
+
+  afterEach: (browser, done) => {
+    screenshot(browser, 'terra-form', done);
+  },
+
+  'Displays a default NumberField with a number input': (browser) => {
+    browser
+      .url(`http://localhost:${browser.globals.webpackDevServerPort}/#/tests/form-tests/number-field/default`)
+      .assert.elementPresent('.terra-Form-field')
+      .assert.elementPresent('input[type="number"]');
+  },
+
+  'Displays a populated NumberField with correct label, help message, and choices': (browser) => {
+    browser
+      .url(`http://localhost:${browser.globals.webpackDevServerPort}/#/tests/form-tests/number-field/populated_uncontrolled`)
+      .assert.elementPresent('.terra-Form-field')
+      .assert.elementPresent('.terra-Form-field--inline')
+      .assert.containsText('label', 'Sales Tax Rate')
+      .assert.containsText('.terra-Form-helpText', "Your county's office may have this information")
+      .assert.containsText('.terra-Form-error', 'This field is required');
+  },
+
+  'Properly initializes NumberField as uncontrolled when passed defaultValue': (browser) => {
```

Comment:
I've removed all the tests that are testing react, and instead made sure that value, onChange, and default can be set correctly. https://github.com/cerner/terra-core/pull/410/commits/de1c2e16ab115ac2b74b100d99eb2f85b4904194